packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -1,47 +1,249 @@
+     * @param _MinimumBridgingDelay the new minimum delay.
+     */
+    function setMinimumBridgingDelay(uint64 _MinimumBridgingDelay) public onlyFromCrossDomainAccount(bridgeRouter) {
+        minimumBridgingDelay = _MinimumBridgingDelay;
+        emit SetMinimumBridgingDelay(minimumBridgingDelay);
+    }
+
+    /**
+     * @notice Enables L1 owner to whitelist a L1 Token <-> L2 Token pair for bridging.
+     * @dev Only callable by the existing bridgeRouter via the optimism cross domain messenger.
+     * @param l1Token Address of the canonical L1 token. This is the token users will receive on Ethereum.
+     * @param l2Token Address of the L2 token representation. This is the token users would deposit on optimism.
+     */
+    function whitelistToken(address l1Token, address l2Token) public onlyFromCrossDomainAccount(bridgeRouter) {
+        whitelistedTokens[l2Token] = WhitelistedToken({ l1Token: l1Token, lastBridgeTime: uint64(getCurrentTime()) });
 
-    function transferL1Owner(address _l1Owner) public onlyFromCrossDomainAccount(l1Owner) {
-        l1Owner = _l1Owner;
+        emit WhitelistToken(l1Token, l2Token, uint64(getCurrentTime()));
     }
 
-    // Depositor functions
+    /**
+     * @notice L1 owner can enable/disable deposits over all whitelisted tokens.
+     * @dev Only callable by the existing bridgeRouter via the optimism cross domain messenger.
+     * @param _depositsEnabled bool to set if the deposit box should accept/reject deposits.
+     */
+    function setEnableDeposits(bool _depositsEnabled) public onlyFromCrossDomainAccount(bridgeRouter) {
+        depositsEnabled = _depositsEnabled;
+        emit DepositsEnabled(_depositsEnabled);
+    }
 
+    /**************************************
+     *         DEPOSITOR FUNCTIONS        *
+     **************************************/
+
+    /**
+     * @notice Called by L2 user to bridge funds between L2 and L1.
+     * @dev Emits the `FundsDeposited` event which relayers listen for as part of the bridging action.
+     * @dev The caller must first approve this contract to spend `amount` of `l2Token`.
+     * @param recipient L1 address that should receive the tokens.
+     * @param l2Token L2 token to deposit.
+     * @param amount How many L2 tokens should be deposited.
+     * @param maxFeePct Max fraction of the total `amount` that the depositor is willing to pay as a fee. scaled by 1e18.
+     */
     function deposit(
         address recipient,
-        address l1Token,
+        address l2Token,
         uint256 amount,
-        uint256 maxFee
-    ) public {}
+        uint256 maxFeePct
+    ) public onlyIfDepositsEnabled() {
+        require(isWhitelistToken(l2Token), "deposit token not whitelisted");
+        require(maxFeePct <= 1e18, "max fee can not be over 1e18");
+
+        emit FundsDeposited(
+            numberOfDeposits, // the current number of deposits acts as a deposit ID (nonce).
```

Comment:
This comment seems unneccessary. If you really need to clarify it why not just rename `numberOfDeposits` to `depositNonce`?

Reply:
I prefer it as `numberOfDeposits` as it's a counter as a variable, rather than `depositNonce`.

Reply:
it might be worthwhile to set an offset for this at the launch of the contract in the case we re-deploy this contract. however, we can always add this at a later point if/when upgrading

Reply:
yeah good point.